XHR-lib.pl

```diff
@@ -0,0 +1,49 @@
+#!/usr/local/bin/perl
+# A pile of XHR related routines
+
+use strict;
+
+our (%in, %gconfig, $root_directory, $remote_user, $current_theme);
+
+sub xhr
+{
+
+my %data    = ();
+my $output  = sub {
+    my ($data) = @_;
+    print "x-no-links: 1\n";
+    print_json($data);
+    };
+
+# Fetch actions
+if ($in{'action'} eq "fetch") {
+    # Download types
+    if ($in{'type'} eq "download") {
+        # Format Blob format
+        if ($in{'subtype'} eq "blob") {
+            # Download using giving URL
+            my $url = $in{'url'};
+            if ($url) {
+                # Unescape possibly HTML escaped
+                # image URL (LinkedIn and other)
+                $url = &html_unescape($url);
+                my ($host, $port, $page, $ssl) = &parse_http_url($url);
+                my $img;
+                &http_download($host, $port, $page, \$img, undef, undef, $ssl, undef, undef, 10);
+                # Get MIME content type
+                eval "use File::MimeInfo";
+                my $img_tmp = &transname();
+                &write_file_contents($img_tmp, $img);
+                my $mime_type = mimetype($img_tmp);
```

Comment:
Can't we get the MIME type back as one of the headers from http_download? That seems safer than guessing it, and we then don't have to write to a temp file..

Reply:
I'll check, not sure off hand.

Reply:
Although, do you mean response headers? Can we really rely on them?

Reply:
> Can't we get the MIME type back as one of the headers from http_download? That seems safer than guessing it, and we then don't have to write to a temp file..

Jamie, check my two last commits. Does it look good to you?

Reply:
Looks better, but why not just always use the returned MIME type? It's guaranteed to be set..